src/models/events/MessageEvent.ts

```diff
@@ -251,4 +251,11 @@ export class MessageEvent<T extends MessageEventContent> extends RoomEvent<T> {
         if (!body && body !== "") throw new EventRedactedError("missing body");
         return body;
     }
+
+    /**
+     * The `external_url` of the message, if it exists
+     */
+    public get externalUrl(): string | undefined {
```

Comment:
This is overly specified - just needs to be a string. 
```suggestion
    public get externalUrl(): string {
```

Reply:
As the external URL is optional that is an inaccurate type, though, it *can*, in fact, be `undefined`

Reply:
In a perfect world, yes. In practice this project documents when falsey values are returned and advertises the preferred type. Otherwise the `body` and `msgtype` above would be over specified as well.

Reply:
both `body` and `msgtype` are **required** parameters, thus their reply type **has** to be `string`. As `external_url` is **optional** it can be `string | undefined`. Different situation

Reply:
They are not required: a redacted event will blow them away.

Reply:
And you through an error if those are empty if they are missing, as the event is redacted.

For a non-redacted event `external_url` isn't required, though, as such it is a different situation.

Reply:
Oh, so we do. This is fine then.